api/src/main/java/com/redhat/ipaas/api/v1/rest/PaginationOptionsFromQueryParams.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.ipaas.api.v1.rest;
+
+import com.redhat.ipaas.api.v1.rest.util.PaginationOptions;
+
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.UriInfo;
+
+
+/**
+ * @author roland
+ * @since 14/12/16
+ */
+class PaginationOptionsFromQueryParams implements PaginationOptions {
+
+    private final int page;
+    private final int perPage;
+
+    PaginationOptionsFromQueryParams(UriInfo uri) {
+        MultivaluedMap<String, String> queryParams = uri.getQueryParameters();
+        String pageQuery = queryParams.getFirst("page");
+        if (pageQuery == null || pageQuery.isEmpty()) {
+            page = 1;
+        } else {
+            page = Integer.parseInt(pageQuery);
```

Comment:
There should be somewhere a check for negative values, otherwise this will break the paging logic. This could be here, in the PaginationFilterExtractor or when doing the pagination.